internal/git/gitsync_processor.go

```diff
@@ -157,8 +185,228 @@ func watchRepo(ctx context.Context, r *git.Repository, gitSync *v1.GitSync, rest
 			return err
 		}
 	} else {
-		// TODO: monitoring with intervals
-		logger.Debug("monitoring with intervals")
+		ticker := time.NewTicker(timeInterval * time.Minute)
+		defer ticker.Stop()
+		for {
+			select {
+			case <-ticker.C:
+				patchedResources, recentHash, err := CheckForRepoUpdates(ctx, r, repo, &gitSync.Status, namespace)
+				if err != nil {
+					return err
+				}
+
+				err = updateCommitStatus(ctx, k8Client, namespacedName, recentHash, repo, logger)
```

Comment:
just note that ultimately we will also want to include errors in the CommitStatus, which is [this](https://github.com/numaproj-labs/numaplane/issues/60) issue. But this is fine for now, and the other call to this function is also not incorporating errors either

Reply:
Yeah we can defer to that issue.